commons-redis/src/main/scala/com/avsystem/commons/redis/actor/ConnectionPoolActor.scala

```diff
@@ -0,0 +1,79 @@
+package com.avsystem.commons
+package redis.actor
+
+import java.util.concurrent.ConcurrentLinkedDeque
+
+import akka.actor.{Actor, ActorRef, Props}
+import com.avsystem.commons.redis.NodeAddress
+import com.avsystem.commons.redis.actor.ConnectionPoolActor._
+import com.avsystem.commons.redis.config.{ConnectionConfig, NodeConfig}
+import com.avsystem.commons.redis.exception.RedisException
+import com.typesafe.scalalogging.LazyLogging
+
+import scala.annotation.tailrec
+
+class ConnectionPoolActor(address: NodeAddress, config: NodeConfig, queue: ConcurrentLinkedDeque[QueuedConn])
```

Comment:
Are there multiple actors using the queue? From the actor's perspective it does not look like it has to be concurrent.

Reply:
Queue is used directly in `RedisNodeClient` which is outside actor context.

Reply:
I know that. I just believe that the actor does not have to enforce this requirement in *his* types, he only cares that it's a queue.